service/mongodb/store.go

```diff
@@ -0,0 +1,154 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/meateam/fav-service/service"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo/readpref"
+)
+
+const (
+	// MongoObjectIDField is the default mongodb unique key.
+	MongoObjectIDField = "_id"
+
+	// FavoriteCollectionName is the name of the favorites collection.
+	FavoriteCollectionName = "favorite"	
+
+	// FavoriteBSONFileIDField is the name of the fileID field in BSON.
+	FavoriteBSONFileIDField = "fileID"
+
+	// FavoriteBSONUserIDField is the name of the userID field in BSON.
+	FavoriteBSONUserIDField = "userID"
+)
+
+// MongoStore holds the mongodb database and implements Store interface.
+type MongoStore struct {
+	DB *mongo.Database
+}
+
+// newMongoStore returns a new store.
+func newMongoStore(db *mongo.Database) (MongoStore, error) {
+	collection := db.Collection(FavoriteCollectionName)
+	indexes := collection.Indexes()
+	indexModel := mongo.IndexModel{
+		Keys: bson.D{
+			bson.E{
+				Key:   FavoriteBSONFileIDField,
+				Value: 1,
+			},
+			bson.E{
+				Key:   FavoriteBSONUserIDField,
+				Value: 1,
+			},
+		},
+		Options: options.Index().SetUnique(true),
+	}
+
+	_, err := indexes.CreateOne(context.Background(), indexModel)
+	if err != nil {
+		return MongoStore{}, err
+	}
+
+	return MongoStore{DB: db}, nil
+}
+
+
+
+// GetAll gets all user favorite files by userID. 
+// If the user doesnt have favorite files at all, it will return empty array. 
+// If successful returns an array of favorite objects. 
+func (s MongoStore) GetAll(ctx context.Context, filter interface{}) ([]primitive.D, error) {
```

Comment:
see my note before of why returning a primitive

Reply:
@maya-fisher

Reply:
Done

Reply:
I don't like the changes. please see the implementation in [permission service ](https://github.com/meateam/permission-service/blob/master/service/mongodb/store.go)

Reply:
Done.